### ISSUE TITLE
Auto-variance calculation doesn't take into account manually-specified variance of base class type parameters

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4713]
The auto-variance calculation implemented for PEP 695 does not currently take into account the case where a new-style generic class derives from an old-style generic class where the variance of a type parameter is manually specified.

In the example below, `T1@A` is invariant, so `T@B` should be inferred to be invariant as well, but currently it is inferred as covariant.

```python
from typing import Generic, TypeVar

T1 = TypeVar("T1")
T_co = TypeVar("T_co", covariant=True)

class A(Generic[T1]):
    pass

class B[T](A[T]):
    ...

def func(a: B[int], b: B[float]):
    # The type parameter for B should be invariant,
    # so both of these assignments should generate
    # an error. Currently, only one of them does.
    if 1 < 2:
        a = b
    else:
        b = a
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.297, which I just published. It will also be included in a future release of pylance.
